tools/twix/src/panels/enum_plot.rs

```diff
@@ -0,0 +1,354 @@
+                    - self.viewport_width * (1.0 - OVERSCAN_FACTOR);
+            }
+            ViewportMode::Free => {}
+        }
+
+        plot_ui.set_plot_bounds(PlotBounds::from_min_max(
+            [self.scroll_pos, 0.0],
+            [self.scroll_pos + self.viewport_width, 1.0],
+        ));
+
+        for segment in self.segments() {
+            Self::plot_segment(plot_ui, &segment);
+        }
+    }
+
+    fn plot(&mut self, ui: &mut Ui) -> Response {
+        let plot = Plot::new("Jürgen")
+            .height(64.0)
+            .show_y(false)
+            .y_axis_width(0)
+            .y_grid_spacer(|_| vec![])
+            .show_grid(false)
+            .allow_scroll(false)
+            .allow_drag(false)
+            .allow_zoom(false)
+            .label_formatter(|_name, plot_point| format!("Message #{}", plot_point.x as isize))
+            .show(ui, |plot_ui| self.show_plot(plot_ui));
+
+        if plot.response.double_clicked() {
+            self.viewport_width = (self.messages_count as f64).max(1.0);
+            self.scroll_pos = 0.0;
+        }
+
+        if let Some(hover_pos) = plot.response.hover_pos() {
+            let plot_hover_pos = plot.transform.value_from_position(hover_pos).x;
+
+            let hovered_segment = self
+                .segments()
+                .iter()
+                .find(|segment| {
+                    (segment.start as f64) < plot_hover_pos && plot_hover_pos < (segment.end as f64)
+                })
+                .cloned();
+
+            if let Some(tooltip) = hovered_segment.and_then(|segment| segment.tooltip()) {
+                show_tooltip_at_pointer(
+                    ui.ctx(),
+                    eframe::egui::Id::new("enum_plot_segment_tooltip"),
+                    |ui| {
+                        ui.label(tooltip);
+                    },
+                );
+            }
+        }
+
+        plot.response
+    }
+
+    fn subscribe(&mut self) {
+        self.change_buffer = match CyclerOutput::from_str(&self.output_key) {
+            Ok(output) => {
+                let buffer = self.nao.subscribe_changes(output);
+                Some(buffer)
+            }
+            Err(error) => {
+                error!("Failed to subscribe: {:#}", error);
+                None
+            }
+        };
+    }
+}
+
+impl Widget for &mut EnumPlotPanel {
+    fn ui(self, ui: &mut Ui) -> Response {
+        let error = self.update_changes().err();
+
+        ui.vertical(|ui| {
+            self.plot(ui);
+
+            if let Some(error) = error {
+                ui.label(RichText::new(error).color(Color32::RED));
+            }
+
+            ui.horizontal_top(|ui| {
+                let subscription_field = ui.add(CompletionEdit::outputs(
+                    &mut self.output_key,
+                    self.nao.as_ref(),
+                ));
+
+                if subscription_field.changed() {
+                    info!("Subscribing: {}", self.output_key);
```

Comment:
is this a useful log message?

Reply:
This is consistent with the plot panel
https://github.com/HULKs/hulk/blob/cef35f22440b860de66d90ee8e74f407441d23f3/tools/twix/src/panels/plot.rs#L119-L122